src/main/java/com/example/java/test/junior/developer/security/KeycloakAuthClient.java

```diff
@@ -0,0 +1,44 @@
+package com.example.java.test.junior.developer.security;
+
+import com.example.java.test.junior.developer.dto.LoginResponseDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ResponseStatusException;
+
+@Component
+@RequiredArgsConstructor
+public class KeycloakAuthClient {
+
+  private final WebClient webClient;
+  private final KeycloakConfiguration keycloakConfiguration;
+
+  public String getAccessToken(String username, String password) {
+    var requestBody = buildRequestBody(username, password);
+    return webClient.post()
+        .uri(keycloakConfiguration.getTokenUri())
+        .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+        .bodyValue(requestBody)
+        .retrieve()
+        .bodyToMono(LoginResponseDto.class)
+        .blockOptional()
+        .orElseThrow(
+            () -> new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid credentials"))
+        .getAccessToken();
```

Comment:
Why you return access token here if you controller should return LoginResponseDto??????????

Reply:
You method should return LoginResponseDto. And your service method should. And controller.